exercises/leap/example_gen.go

```diff
@@ -15,7 +16,8 @@ func main() {
 		log.Fatal(err)
 	}
 	var j js
-	if err := gen.Gen("leap.json", &j, t); err != nil {
+	f := path.Join("exercises", "leap", "canonical-data.json")
```

Comment:
Consider the effect of making this change across every single generator - we will be typing `path.Join("exercises", whateverSlug, "canonical-data.json")` a lot. Can/should this line be moved into the `Gen` function, and we just pass `leap` into `Gen`?

I won't oppose this change for this PR, but I'd advise it if working on more generators.